### PR TITLE
MDTZ-1232: Write to the `jira_associated_figma_design.input_url` column (PR 2/2)

### PR DESCRIPTION
**Pull request by akostevich-atlassian**
Please, review and merge https://github.com/atlassian-labs/figma-for-jira/pull/190 first.

## Changes

- **feat:** Adds `AssociatedFigmaDesign.inputUrl` to the domain model.
- **feat:** Stores `AssociatedFigmaDesign.inputUrl` (in the `jira_associated_figma_design.input_url` column)

### CODE REVIEW

File: src/usecases/backfill-design-use-case.ts [hunk 1]
```diff
@@ -85,6 +85,7 @@ export const backfillDesignUseCase = {
 			await associatedFigmaDesignRepository.upsert({
 				designId: figmaDesignId,
 				associatedWithAri: associateWith.ari,
+				inputUrl: designUrl,
```
Comment (jvrankul23) [inline src/usecases/backfill-design-use-case.ts:88]: @akostevich-atlassian should we also modify the associate entity use case?

File: src/usecases/backfill-design-use-case.ts [hunk 2]
```diff
@@ -86,6 +86,7 @@ export const backfillDesignUseCase = {
 				designId: figmaDesignId,
 				associatedWithAri: associateWith.ari,
 				connectInstallationId: connectInstallation.id,
+				inputUrl: designUrl,
```
Comment (akostevich-atlassian) [inline src/usecases/backfill-design-use-case.ts:89]: **note:** Writing a new column within the "Backfill" use case.

File: src/usecases/associate-design-use-case.ts [hunk 3]
```diff
@@ -82,6 +82,7 @@ export const associateDesignUseCase = {
 				designId: figmaDesignId,
 				associatedWithAri: associateWith.ari,
 				connectInstallationId: connectInstallation.id,
+				inputUrl: designUrl,
```
Comment (akostevich-atlassian) [inline src/usecases/associate-design-use-case.ts:85]: **note:** Writing a new column within the "Associate" use case.

File: prisma/schema.prisma [hunk 4]
```diff
@@ -22,10 +22,11 @@ model ConnectInstallation {
...
 model AssociatedFigmaDesign {
-  id                    BigInt              @id @default(autoincrement())
-  fileKey               String              @map("file_key")
-  nodeId                String              @map("node_id") // Prisma does not support nullable columns with the composite indexes: https://github.com/prisma/prisma/issues/3197
-  associatedWithAri     String              @map("associated-with-ari")
+  id                BigInt  @id @default(autoincrement())
+  fileKey           String  @map("file_key")
+  nodeId            String  @map("node_id") // Prisma does not support nullable columns with the composite indexes: https://github.com/prisma/prisma/issues/3197
+  associatedWithAri String  @map("associated-with-ari")
+  inputUrl          String? @map("input_url")
```
Comment (akostevich-atlassian) [inline prisma/schema.prisma:29]: **note:** Adding a new column.

If you see more changes in this file, please, click "Hide whitespace" on the toolbar. It seems Prettier reformatted this file a bit.

File: src/usecases/backfill-design-use-case.ts (hunk 1 above)
Comment (akostevich-atlassian) [reply src/usecases/backfill-design-use-case.ts]: Great catch, Jon. We absolutely should!

**Comment by elainefigma** [comment]
Overall, this approach looks good to me! A minor request - could we split out the code that performs the database migration and the code that uses a new column? Then, could we first merge in the prisma/ folder changes? 

For some context - we updated the figma-for-jira deploy pipeline so that we don't run database migrations as a pre-requisite before starting the app. That way, if we make some underlying database change, then we won't break the figma-for-jira deploy process. 

It'll be easier for us to deploy this change if we:
- first sync the change with the prisma/ folder changes. run the database migration.
- separately sync the change that use the new column. deploy the new app code.

**Comment by akostevich-atlassian** [comment]
@elainefigma , yes, sure thing. I will split the PR into 2. Thank you for letting me know!

**Comment by akostevich-atlassian** [comment]
@elainefigma , I split the changes into the 2 PRs:

- https://github.com/atlassian-labs/figma-for-jira/pull/187 (current)
- https://github.com/atlassian-labs/figma-for-jira/pull/190

**Pull request merged**